grails-app/controllers/org/pih/warehouse/reporting/ReportController.groovy

```diff
@@ -692,9 +692,11 @@ class ReportController {
                 def qtyOnHand = items.qtyOnHand.sum() ?: 0
                 def qtyExpiring = productExpiry.collect {it.quantity_on_hand}.sum() ?: 0
                 def qtyAvailable = qtyOnOrder + qtyOnHand - qtyExpiring ?: 0
-                def avgDemand = productExpiry.collect {it.average_daily_demand}.size() > 0 ? productExpiry.collect {it.average_daily_demand}.get(0).setScale(1, RoundingMode.HALF_UP) : 0
+                def avgDemand = productExpiry.collect {it.average_daily_demand}.size() > 0 ? productExpiry.collect {it.average_daily_demand}.get(0).setScale(1, RoundingMode.HALF_UP)*30 : 0
                 def monthsOfStock = ((replenishmentPeriodDays.toInteger() + leadTimeDays.toInteger())/30).setScale(1, RoundingMode.HALF_UP) ?: 0
                 def qtyNeeded = avgDemand * monthsOfStock ?: 0
+                def qtyToOrder = BigDecimal.ZERO.max(qtyNeeded - qtyOnHand)
+                def unitPrice = Product.findByProductCode(items[0].productCode).pricePerUnit ?: 0
```

Comment:
You have `...{ productId, items ->...` in the collect, use that instead of `items[0].productCode`

Reply:
corrected